notebooks/Liquidators_data.ipynb

```diff
@@ -0,0 +1,977 @@
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Processing zklend from Google Storage...\n",
+      "Processing nostra_alpha from Google Storage...\n",
+      "Processing nostra_mainnet from Google Storage...\n",
+      "Processing hashstack_v0 from Google Storage...\n",
+      "Processing hashstack_v1 from Google Storage...\n",
+      "Combined dataframe shape: (305033, 11)\n"
+     ]
+    }
+   ],
+   "source": [
+    "def load_data_from_google_storage(url):\n",
+    "    \"\"\"Load data from a Google Storage URL using pandas.read_parquet.\"\"\"\n",
+    "    response = requests.get(url)\n",
+    "    response.raise_for_status()  # Ensure the request was successful\n",
+    "\n",
+    "    with BytesIO(response.content) as f:\n",
+    "        df = pd.read_parquet(f)\n",
+    "    return df\n",
+    "\n",
+    "def load_data_from_local_parquet(file_path):\n",
+    "    \"\"\"Load data from a local Parquet file using pandas.read_parquet.\"\"\"\n",
+    "    df = pd.read_parquet(file_path)\n",
+    "    return df\n",
+    "\n",
+    "def load_protocol_data(source, config):\n",
+    "    \"\"\"Load data based on the specified source.\"\"\"\n",
+    "    combined_protocols_df = pd.DataFrame()\n",
+    "    if source == 'google_storage':\n",
+    "        for protocol, url in config.items():\n",
+    "            print(f\"Processing {protocol} from Google Storage...\")\n",
+    "            df_protocol = load_data_from_google_storage(url)\n",
+    "            df_protocol['Protocol'] = protocol\n",
+    "            combined_protocols_df = pd.concat([combined_protocols_df, df_protocol], ignore_index=True)\n",
+    "    elif source == 'local_parquet':\n",
+    "        for protocol, file_path in config.items():\n",
+    "            print(f\"Processing {protocol} from local Parquet file...\")\n",
+    "            df_protocol = load_data_from_local_parquet(file_path)\n",
+    "            df_protocol['Protocol'] = protocol\n",
+    "            combined_protocols_df = pd.concat([combined_protocols_df, df_protocol], ignore_index=True)\n",
+    "    return combined_protocols_df\n",
+    "\n",
+    "# Configuration for Google Storage\n",
+    "google_storage_config = {\n",
+    "    'zklend': 'https://storage.googleapis.com/derisk-persistent-state/zklend_data/events_sample.parquet',\n",
+    "    'nostra_alpha': 'https://storage.googleapis.com/derisk-persistent-state/nostra_alpha_data/events_sample.parquet',\n",
+    "    'nostra_mainnet': 'https://storage.googleapis.com/derisk-persistent-state/nostra_mainnet_data/events_sample.parquet',\n",
+    "    'hashstack_v0': 'https://storage.googleapis.com/derisk-persistent-state/hashstack_v0_data/events_sample.parquet',\n",
+    "    'hashstack_v1': 'https://storage.googleapis.com/derisk-persistent-state/hashstack_v1_data/events_sample.parquet'\n",
+    "}\n",
+    "\n",
+    "# Configuration for local Parquet files\n",
+    "local_parquet_config = {\n",
```

Comment:
Why is this needed? Does the loading from GCS fail sometimes?

Reply:
In the decription of the issue it was mentioned that the code should be able to accomodate reading from local files as well. Hence I had used that. I have now updated the code now.

Reply:
Oh, I see. But the description says `local database`, meaning local Postgres db, for example.

Reply:
Ah, I misunderstood that earlier. Please let me know if the updated code looks ok. Need your help to understand if I interpreted correctly now.

Reply:
Yes, it looks good now.